From 2f0a0c00627d23eb6dc8809794a6280480eac7ef Mon Sep 17 00:00:00 2001
From: Aliya Rahmani <aliyarahmani786@gmail.com>
Date: Sun, 18 Oct 2020 19:10:43 +0000
Subject: [PATCH 10/18] Simplified the language in instructions

---
 help/C/timer.page | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/help/C/timer.page b/help/C/timer.page
index 6669507..ff18790 100644
--- a/help/C/timer.page
+++ b/help/C/timer.page
@@ -26,7 +26,7 @@
 
   <title>Set the timer</title>
 
-  <p>Set the time for a second by second countdown to zero. You will be
+  <p>Set the amount of time for the timer countdown. You will be
   notified when the countdown has finished.</p>
 
   <steps>
-- 
2.29.2

