From cbf843094121767864d183d537694c6543b6018f Mon Sep 17 00:00:00 2001
From: Julian Sparber <julian@sparber.net>
Date: Thu, 12 Nov 2020 16:22:27 +0100
Subject: [PATCH 17/18] alarm-face: Save active changes to gsettings

---
 src/alarm-face.vala | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/alarm-face.vala b/src/alarm-face.vala
index bf0d387..995143b 100644
--- a/src/alarm-face.vala
+++ b/src/alarm-face.vala
@@ -66,6 +66,7 @@ public class Face : Gtk.Stack, Clocks.Clock {
         });
 
         listbox.bind_model (alarms, (item) => {
+            item.notify["active"].connect (save);
             return new Row ((Item) item, this);
         });
 
-- 
2.29.2

